### DISCUSSION TITLE
Resolve security vulnerabilities in Docker containers of Prefect server

### DISCUSSION THREAD

**Discussion by niakki** [PrefectHQ/prefect#4180] [Q&A]
Hello,
We are evaluating prefect core to replace our workflow management. As part of security protocol all container images are scanned which included all the 5 docker containers of prefect core (server, Apollo, ui, Hasura/graphql & Postgres:11). All of these containers seem to have vulnerabilities and most of them are medium to low.
However, the ones which are critical & high needs to be resolved before we can use them for prod, based on our observation these packages are system libs (like linux kernel, shadow, glibc etc. something which prefect doesn’t directly use). Is it possible to update the base image to say Ubuntu 20.4 (in our tests between Debian and ubuntu, the latter seems to have just about 16 [1 medium rest all low vulnerabilities])? 

Screen shots -

![Screen Shot 2021-02-26 at 17 01 29](https://user-images.githubusercontent.com/36913184/109295575-f9e43280-7854-11eb-9396-be67e548c8b7.png)
![Screen Shot 2021-02-26 at 17 01 44](https://user-images.githubusercontent.com/36913184/109295580-fc468c80-7854-11eb-98a8-0a88dfff7057.png)
![Screen Shot 2021-02-26 at 16 15 05](https://user-images.githubusercontent.com/36913184/109295583-fd77b980-7854-11eb-84ee-5010c999add8.png)

_PS - Scans have been made using GCP’s vulnerability scan service in GCR_

Any other recommended approach we could follow to over come this issue is greatly appreciated. :slightly_smiling_face:

[Slack reference](https://prefect-community.slack.com/archives/CL09KU1K7/p1614269132285700?thread_ts=1614260765.274600&cid=CL09KU1K7)

**Comment by znicholasbrown** [comment]
Posting here for anyone that stumbles across the discussion and not the issue: https://github.com/PrefectHQ/server/issues/197#issuecomment-786775782
